client/app/bundles/course/material/folders/handles.ts

```diff
@@ -7,17 +7,26 @@ const getFolderTitle = async (
   courseUrl: string,
   folderId: number,
 ): Promise<CrumbPath> => {
-  const { data } = await CourseAPI.folders.fetch(folderId);
+  try {
+    const { data } = await CourseAPI.folders.fetch(folderId);
+    const workbinUrl = `${courseUrl}/materials/folders/${data.breadcrumbs[0].id}`;
 
-  const workbinUrl = `${courseUrl}/materials/folders/${data.breadcrumbs[0].id}`;
-
-  return {
-    activePath: workbinUrl,
-    content: data.breadcrumbs.map((crumb) => ({
-      title: crumb.name,
-      url: `materials/folders/${crumb.id}`,
-    })),
-  };
+    return {
+      activePath: workbinUrl,
+      content: data.breadcrumbs.map((crumb) => ({
+        title: crumb.name,
+        url: `materials/folders/${crumb.id}`,
+      })),
+    };
+  } catch (error) {
+    return {
+      activePath: courseUrl,
+      content: {
+        title: 'Folder Not Found',
+        url: courseUrl,
```

Comment:
it should be pointing to `{courseTitle}/Materials`. Please refer to the breadcrumb generated when the folder exists (you can refer to the comparison between breadcrumbs when file exist vs. file not exist, this one should reflect on that as well)

Also, please be careful that Materials here is actually because the Sidebar component's name is `Material`. This can be changed through `Course Settings > Materials`. Please be aware of this when you make the change for this

Reply:
Since the system doesn't handle the base for `/courses/2/materials/folders` or `/courses/2/materials`, not sure what's the good choice for exceptional case here.
<img width="730" alt="Screenshot 2024-10-24 at 10 09 34 AM" src="https://github.com/user-attachments/assets/6d558663-6017-492a-b574-391710eac234">
<img width="686" alt="Screenshot 2024-10-24 at 10 24 31 AM" src="https://github.com/user-attachments/assets/36b7e93c-452e-4a94-9d72-bac2ea90498a">

Reply:
You could make a route `courses/<course-id>/materials` redirect to the default root materials folder, otherwise you can check how the sidebar gets the default root materials folder link.

Reply:
I hope this will help, trying to apply it
![Uploading Screenshot 2024-10-24 at 1.50.50 PM.png…]()

Reply:
wrong. That function `getCurrFolderInfo` relies on the info from the backend regarding the current folder, should it exist. How if I directly go into `courses/<course-id>/materials/folder/<some-non-existing-folder-id>` instead of clicking through `Materials` sidebar?

Reply:
I have successfully updated the workbin URL, and its functionality now mirrors the sidebar, just as @cysjonathan suggested.